File: lib/pb/groupshare/data/foo/foo.op.dart
import 'package:protod/delta/delta.dart' as delta;
import 'package:protod/delta/delta.pb.dart' as delta;
import 'package:fixnum/fixnum.dart' as fixnum;
import 'package:groupshare/pb/groupshare/data/foo/foo.pb.dart' as pb;

Op_root_type Op() {
  return Op_root_type();
}

class Op_root_type {
  Op_root_type();
  Foo_type Foo() {
    return Foo_type([]);
  }
}

class Foo_type extends delta.Location {
  Foo_type(List<delta.Locator> location) : super(location);
  delta.String_scalar Bar() {
    return delta.String_scalar(delta.copyAndAppendField(location, "bar", 1));
  }
  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(pb.Foo value) {
    return delta.set(location, value);
  }

}

class Foo_type_list extends delta.Location {
  Foo_type_list(List<delta.Locator> location) : super(location);
  Foo_type Index(int i) {
    return Foo_type(delta.copyAndAppendIndex(location, fixnum.Int64(i)));
  }
  delta.Op Insert(int index, pb.Foo value) {
    return delta.insert(delta.copyAndAppendIndex(location, fixnum.Int64(index)), value);
  }

  delta.Op Move(int from, int to) {
    return delta.move(delta.copyAndAppendIndex(location, fixnum.Int64(from)), fixnum.Int64(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(List<pb.Foo> value) {
    return delta.set(location, value);
  }

}

class Foo_type_bool_map extends delta.Location {
  Foo_type_bool_map(List<delta.Locator> location) : super(location);
  Foo_type Key(bool key) {
    return Foo_type(delta.copyAndAppendKeyBool(location, key));
  }
  delta.Op Rename(bool from, bool to) {
    return delta.rename(delta.copyAndAppendKeyBool(location, from), delta.keyBool(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<bool, pb.Foo> value) {
    return delta.set(location, value);
  }

}

class Foo_type_int32_map extends delta.Location {
  Foo_type_int32_map(List<delta.Locator> location) : super(location);
  Foo_type Key(int key) {
    return Foo_type(delta.copyAndAppendKeyInt32(location, key));
  }
  delta.Op Rename(int from, int to) {
    return delta.rename(delta.copyAndAppendKeyInt32(location, from), delta.keyInt32(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<int, pb.Foo> value) {
    return delta.set(location, value);
  }

}

class Foo_type_int64_map extends delta.Location {
  Foo_type_int64_map(List<delta.Locator> location) : super(location);
  Foo_type Key(int key) {
    return Foo_type(delta.copyAndAppendKeyInt64(location, fixnum.Int64(key)));
  }
  delta.Op Rename(int from, int to) {
    return delta.rename(delta.copyAndAppendKeyInt64(location, fixnum.Int64(from)), delta.keyInt64(fixnum.Int64(to)));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<fixnum.Int64, pb.Foo> value) {
    return delta.set(location, value);
  }

}

class Foo_type_uint32_map extends delta.Location {
  Foo_type_uint32_map(List<delta.Locator> location) : super(location);
  Foo_type Key(int key) {
    return Foo_type(delta.copyAndAppendKeyUint32(location, key));
  }
  delta.Op Rename(int from, int to) {
    return delta.rename(delta.copyAndAppendKeyUint32(location, from), delta.keyUint32(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<int, pb.Foo> value) {
    return delta.set(location, value);
  }

}

class Foo_type_uint64_map extends delta.Location {
  Foo_type_uint64_map(List<delta.Locator> location) : super(location);
  Foo_type Key(int key) {
    return Foo_type(delta.copyAndAppendKeyUint64(location, fixnum.Int64(key)));
  }
  delta.Op Rename(int from, int to) {
    return delta.rename(delta.copyAndAppendKeyUint64(location, fixnum.Int64(from)), delta.keyUint64(fixnum.Int64(to)));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<fixnum.Int64, pb.Foo> value) {
    return delta.set(location, value);
  }

}

class Foo_type_string_map extends delta.Location {
  Foo_type_string_map(List<delta.Locator> location) : super(location);
  Foo_type Key(String key) {
    return Foo_type(delta.copyAndAppendKeyString(location, key));
  }
  delta.Op Rename(String from, String to) {
    return delta.rename(delta.copyAndAppendKeyString(location, from), delta.keyString(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<String, pb.Foo> value) {
    return delta.set(location, value);
  }

}

File: server/pb/groupshare/data/foo/foo.op.go
package foo

import "github.com/dave/protod/delta"

func Op() Op_root_type {
	return Op_root_type{}
}

type Op_root_type struct{}

func (Op_root_type) Foo() Foo_type {
	return Foo_type{}
}

type Foo_type struct {
	location []*delta.Locator
}

func (b Foo_type) Location_get() []*delta.Locator {
	return b.location
}
func NewFoo_type(l []*delta.Locator) Foo_type {
	return Foo_type{location: l}
}
func (b Foo_type) Bar() delta.String_scalar {
	return delta.NewString_scalar(delta.CopyAndAppendField(b.location, "bar", 1))
}
func (b Foo_type) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Foo_type) Set(value *Foo) *delta.Op {
	return delta.Set(b.location, value)
}

type Foo_type_list struct {
	location []*delta.Locator
}

func (b Foo_type_list) Location_get() []*delta.Locator {
	return b.location
}
func NewFoo_type_list(l []*delta.Locator) Foo_type_list {
	return Foo_type_list{location: l}
}
func (b Foo_type_list) Index(i int) Foo_type {
	return NewFoo_type(delta.CopyAndAppendIndex(b.location, int64(i)))
}
func (b Foo_type_list) Insert(index int, value *Foo) *delta.Op {
	return delta.Insert(delta.CopyAndAppendIndex(b.location, int64(index)), value)
}
func (b Foo_type_list) Move(from, to int) *delta.Op {
	return delta.Move(delta.CopyAndAppendIndex(b.location, int64(from)), int64(to))
}
func (b Foo_type_list) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Foo_type_list) Set(value []*Foo) *delta.Op {
	return delta.Set(b.location, value)
}

type Foo_type_bool_map struct {
	location []*delta.Locator
}

func (b Foo_type_bool_map) Location_get() []*delta.Locator {
	return b.location
}
func NewFoo_type_bool_map(l []*delta.Locator) Foo_type_bool_map {
	return Foo_type_bool_map{location: l}
}
func (b Foo_type_bool_map) Key(key bool) Foo_type {
	return NewFoo_type(delta.CopyAndAppendKeyBool(b.location, key))
}
func (b Foo_type_bool_map) Rename(from, to bool) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyBool(b.location, from), to)
}
func (b Foo_type_bool_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Foo_type_bool_map) Set(value map[bool]*Foo) *delta.Op {
	return delta.Set(b.location, value)
}

type Foo_type_int32_map struct {
	location []*delta.Locator
}

func (b Foo_type_int32_map) Location_get() []*delta.Locator {
	return b.location
}
func NewFoo_type_int32_map(l []*delta.Locator) Foo_type_int32_map {
	return Foo_type_int32_map{location: l}
}
func (b Foo_type_int32_map) Key(key int) Foo_type {
	return NewFoo_type(delta.CopyAndAppendKeyInt32(b.location, int32(key)))
}
func (b Foo_type_int32_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyInt32(b.location, int32(from)), int32(to))
}
func (b Foo_type_int32_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Foo_type_int32_map) Set(value map[int32]*Foo) *delta.Op {
	return delta.Set(b.location, value)
}

type Foo_type_int64_map struct {
	location []*delta.Locator
}

func (b Foo_type_int64_map) Location_get() []*delta.Locator {
	return b.location
}
func NewFoo_type_int64_map(l []*delta.Locator) Foo_type_int64_map {
	return Foo_type_int64_map{location: l}
}
func (b Foo_type_int64_map) Key(key int) Foo_type {
	return NewFoo_type(delta.CopyAndAppendKeyInt64(b.location, int64(key)))
}
func (b Foo_type_int64_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyInt64(b.location, int64(from)), int64(to))
}
func (b Foo_type_int64_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Foo_type_int64_map) Set(value map[int64]*Foo) *delta.Op {
	return delta.Set(b.location, value)
}

type Foo_type_uint32_map struct {
	location []*delta.Locator
}

func (b Foo_type_uint32_map) Location_get() []*delta.Locator {
	return b.location
}
func NewFoo_type_uint32_map(l []*delta.Locator) Foo_type_uint32_map {
	return Foo_type_uint32_map{location: l}
}
func (b Foo_type_uint32_map) Key(key int) Foo_type {
	return NewFoo_type(delta.CopyAndAppendKeyUint32(b.location, uint32(key)))
}
func (b Foo_type_uint32_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyUint32(b.location, uint32(from)), uint32(to))
}
func (b Foo_type_uint32_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Foo_type_uint32_map) Set(value map[uint32]*Foo) *delta.Op {
	return delta.Set(b.location, value)
}

type Foo_type_uint64_map struct {
	location []*delta.Locator
}

func (b Foo_type_uint64_map) Location_get() []*delta.Locator {
	return b.location
}
func NewFoo_type_uint64_map(l []*delta.Locator) Foo_type_uint64_map {
	return Foo_type_uint64_map{location: l}
}
func (b Foo_type_uint64_map) Key(key int) Foo_type {
	return NewFoo_type(delta.CopyAndAppendKeyUint64(b.location, uint64(key)))
}
func (b Foo_type_uint64_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyUint64(b.location, uint64(from)), uint64(to))
}
func (b Foo_type_uint64_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Foo_type_uint64_map) Set(value map[uint64]*Foo) *delta.Op {
	return delta.Set(b.location, value)
}

type Foo_type_string_map struct {
	location []*delta.Locator
}

func (b Foo_type_string_map) Location_get() []*delta.Locator {
	return b.location
}
func NewFoo_type_string_map(l []*delta.Locator) Foo_type_string_map {
	return Foo_type_string_map{location: l}
}
func (b Foo_type_string_map) Key(key string) Foo_type {
	return NewFoo_type(delta.CopyAndAppendKeyString(b.location, key))
}
func (b Foo_type_string_map) Rename(from, to string) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyString(b.location, from), to)
}
func (b Foo_type_string_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Foo_type_string_map) Set(value map[string]*Foo) *delta.Op {
	return delta.Set(b.location, value)
}